src/Form/Expression/ConditionContainerForm.php

```diff
@@ -24,65 +25,356 @@ class ConditionContainerForm implements ExpressionFormInterface {
   protected $conditionContainer;
 
   /**
+   * The rules ui handler.
+   *
+   * @var \Drupal\rules\Ui\RulesUiHandlerInterface|null
+   */
+  protected $rulesUiHandler;
+
+  /**
+   * The rules component.
+   *
+   * @var \Drupal\rules\Engine\RulesComponent
+   */
+  protected $component;
+
+  /**
+   * The rule as an expression.
+   *
+   * @var \Drupal\rules\Engine\ExpressionInterface
+   */
+  protected $ruleExpression;
+
+  /**
    * Creates a new object of this class.
    */
   public function __construct(ConditionExpressionContainerInterface $condition_container) {
     $this->conditionContainer = $condition_container;
+    $this->rulesUiHandler = $this->getRulesUiHandler();
+    $this->component = $this->rulesUiHandler->getComponent();
+    $this->ruleExpression = $this->component->getExpression();
   }
 
   /**
    * {@inheritdoc}
    */
-  public function form(array $form, FormStateInterface $form_state) {
-    $form['conditions'] = [
-      '#type' => 'container',
-    ];
-
-    $form['conditions']['table'] = [
-      '#theme' => 'table',
-      '#caption' => $this->t('Conditions'),
-      '#header' => [$this->t('Elements'), $this->t('Operations')],
-      '#empty' => t('None'),
-    ];
+  public function form(array $form, FormStateInterface $form_state, $options = []) {
+    // By default skip.
+    if (!empty($options['init'])) {
+      /* @var $config \Drupal\rules\Entity\ReactionRuleConfig */
+      $config = $this->rulesUiHandler->getConfig();
+      $form['init_help'] = array(
+        '#type' => 'container',
+        '#id' => 'rules-plugin-add-help',
+        'content' => array(
+          '#markup' => t('You are about to add a new @plugin to the @config-plugin %label. Use indentation to make conditions a part of this logic group. See <a href="@url">the online documentation</a> for more information on condition sets.',
+            array(
+              '@plugin' => $this->conditionContainer->getLabel(),
+              '@config-plugin' => $config->bundle(),
+              '%label' => $config->label(),
+              '@url' => 'http://drupal.org/node/1300034',
```

Comment:
Good to know about the newer placeholder. I assume we are talking about the @url element? Since its not user provided input I wouldn't normally worry about XSS escaping for a hard coded url, but I can change it to use the new placeholder quick enough that its a moot point.